app/controllers/katello/api/rhsm/candlepin_proxies_controller.rb

```diff
@@ -108,14 +110,25 @@ def rhsm_index
       respond_for_index :collection => @all_environments
     end
 
+    #api :POST, "/hypervisors/OWNER"
+    # Note that this request comes in as content-type 'text/plain' so that
+    # tomcat won't parse the json.  Here we just pass the plain body through to candlepin
+    def async_hypervisors_update
+      raw_json = request.body.string
+      task = Katello::Resources::Candlepin::Consumer.async_hypervisors(params[:owner], raw_json)
+      async_task(::Actions::Katello::Host::Hypervisors, :task_id => task['id'])
+
+      render :json => task
+    end
+
     #api :POST, "/hypervisors", N_("Update the hypervisors information for environment")
     #desc 'See virt-who tool for more details.'
     def hypervisors_update
       login = User.consumer? ? User.anonymous_api_admin.login : User.current.login
       task = User.as(login) do
         params['owner'] = @organization.label #override owner label if
         params['env'] = nil #hypervisors don't need an environment
-        sync_task(::Actions::Katello::Host::Hypervisors, params.except(:controller, :action, :format))
+        sync_task(::Actions::Katello::Host::Hypervisors, :hypervisor_params => params.except(:controller, :action, :format))
```

Comment:
is this necessary? foreman_virt_who_configure plugin subscribes to this action and uses its inputs